### PR TITLE
Feature/con 26718 automatic right sizing put call

### PR DESCRIPTION
**Pull request by shirbur**

### CODE REVIEW

**Review by dahanmoshe1991 — APPROVED** [review]

File: api/services/ocean/rightsizing/schemas/oceanRightsizingRuleRecommendationApplicationHPA.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+type: object
+title: Ocean Rightsizing Rule Recommendation Application For HPA
+description: >
+  Determines by the rule if recommendation application is allowed for workloads with HPA definition.
+properties:
+  allowHPARecommendation:
```
Comment (anuragsharma-123) [inline api/services/ocean/rightsizing/schemas/oceanRightsizingRuleRecommendationApplicationHPA.yaml:6]: The correct field name in the spotinst-api code is "allowHPARecommendations" with 's' in the end.

File: api/services/ocean/rightsizing/schemas/oceanRightsizingRuleRequest.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+description: >
+  Determines the Ocean Rightsizing rule
+required:
+  - recommendationApplicationIntervals
+properties:
+  ruleName:
+    type: string
+    description: The unique name of the rule
+    example: "dev-rule"
+  restartPods:
```
Comment (anuragsharma-123) [inline api/services/ocean/rightsizing/schemas/oceanRightsizingRuleRequest.yaml:12]: restartPods field is obsolete. Should be removed.

File: api/services/ocean/rightsizing/schemas/oceanRightsizingUpdateRuleRequest.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+type: object
+title: Ocean Rightsizing Rule
+description: >
+  Determines the Ocean Rightsizing rule
+required:
+  - recommendationApplicationIntervals
+properties:
+  restartPods:
```
Comment (anuragsharma-123) [inline api/services/ocean/rightsizing/schemas/oceanRightsizingUpdateRuleRequest.yaml:8]: restartPods field is obsolete. Should be removed.

Comment (IdanShohamNetApp) [reply api/services/ocean/rightsizing/schemas/oceanRightsizingUpdateRuleRequest.yaml]: fixed

File: api/services/ocean/rightsizing/schemas/oceanRightsizingRuleRecommendationApplicationHPA.yaml (hunk 1 above)
Comment (IdanShohamNetApp) [reply api/services/ocean/rightsizing/schemas/oceanRightsizingRuleRecommendationApplicationHPA.yaml]: fixed

**Review by dahanmoshe1991 — APPROVED** [review]

**Review by oded7hoffman — APPROVED** [review]

**Review by oded7hoffman — APPROVED** [review]

**Pull request merged**